onlineconf.go

```diff
@@ -8,20 +8,61 @@ import (
 	"fmt"
 	"io"
 	"log"
+	"os"
+	"path"
 	"strconv"
 	"sync"
+	"syscall"
 
+	"github.com/colinmarc/cdb"
 	"github.com/fsnotify/fsnotify"
-	"github.com/jbarham/go-cdb"
+	"github.com/grandecola/mmap"
```

Comment:
I took a look into this module. The design is not the best. It implements only ReaderAt not Closer. I'd rather prefer to use https://github.com/golang/exp/blob/master/mmap but _exp_ as a module it too huge with extra useless dependencies.
What do you think about forking it into https://github.com/my-mail-ru/exp and then adding module definition into _mmap_ directly? It will simplify `reopen()` and allow to not store `mmapedFile` explicitly.

Reply:
I like exp/mmap more too. Thank you!

However I don't want to fork it. I want to belive (not sure for 100%) removing useless code from binary is a job for compiler but not a reason for forking. Also I don't know any simple (possibly automated) way to backport upstream changes.

I don't consider relying on finalyzer is a good idea. There are [some points](https://crawshaw.io/blog/tragedy-of-finalizers) against finalyzers.

I replaced grandecola/mmap with exp/mmap but I'd like to leave explicit unmmapping.

Reply:
Oh, now I see what you meant about cdb Closer!

But I don't believe that [this](https://github.com/colinmarc/cdb/blob/master/cdb.go#L107) hack is in common for go. I don't see any reason to call `cdb.Close()` because i didn't pass there any `Closer`. `New(...)` method requires only `ReaderAt`.

In my opinion `cdb.Close()` would be implicit and ambigous

I created mmapped file in one place. And I want to call `Close()` on the same object. Not on the other one. 

And I can't see any advantages of `cdb.Close()`. Is there any reason to use implicit `cdb.Close()`?

Reply:
Passing an ownership of a resource is a quite common pattern. Especially it is widely used for the memory management (not in go of course).

cdb module itself doesn't require Closer in New because it doesn't use Close for something reasonable except for the passthrough. Close completely optional and its requirement depends on a used reader.

The choice of cdb.Close() vs mmap.Close() here is fully opinionated. Someone may prefer to pass the ownership below an abstraction and forget about it another one may prefer to highlight an underling implementation. So leave it as you feel right.